src/main/java/randoop/main/GenTests.java

```diff
@@ -1335,4 +1358,150 @@ private Path getResourceDirectoryPath(String resourceDirectory) {
+      throw new RandoopUsageError("Can`t read methods from " + methodlist.toString());
+    }
+    return classnames;
+  }
+
+  /**
+   * Tests whether a method should be omitted. Returns true if the signature matches any pattern in
+   * omit_method list, false otherwise.
+   *
+   * @param signature signature of method to test
+   * @return true if method should be ommited
+   */
+  private static boolean shouldOmitMethod(String signature) {
+    for (Pattern pattern : omit_methods) {
+      if (pattern.matcher(signature).find()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Adds parameter types of method to collection if method is accessible by accessibility predicate
+   * and parameter type is not String, primitive, should not be omitted and is accessible by
+   * accessibility predicate
+   *
+   * @param method method
+   * @param classnames collection to add parameter types to
+   * @param accessibilityPredicate accessibility predicate
+   */
+  private static void addMethodParameterTypesIfShould(
+      Method method,
+      Collection<@ClassGetName String> classnames,
+      AccessibilityPredicate accessibilityPredicate) {
+    if (accessibilityPredicate.isAccessible(method)) {
+      addExecutableParameterTypesIfShould(method, classnames, accessibilityPredicate);
+    }
+  }
+
+  /**
+   * Adds parameter types of constructor to collection if constructor is accessible by accessibility
+   * predicate and parameter type is not String, primitive, should not be omitted and is accessible
+   * by accessibility predicate
+   *
+   * @param constructor constructor
+   * @param classnames collection to add parameter types to
+   * @param accessibilityPredicate accessibility predicate
+   */
+  private static void addConstructorParameterTypesIfShould(
+      Constructor<?> constructor,
+      Collection<@ClassGetName String> classnames,
+      AccessibilityPredicate accessibilityPredicate) {
+    if (accessibilityPredicate.isAccessible(constructor)) {
+      addExecutableParameterTypesIfShould(constructor, classnames, accessibilityPredicate);
+    }
+  }
+
+  /**
+   * Adds parameter types of executable to collection if parameter type is not String, primitive,
+   * should not be omitted and is accessible by accessibility predicate.
+   *
+   * @param executable executable
+   * @param classnames collection to add parameter types to
+   * @param accessibilityPredicate accessibility predicate
+   */
+  private static void addExecutableParameterTypesIfShould(
+      Executable executable,
+      Collection<@ClassGetName String> classnames,
+      AccessibilityPredicate accessibilityPredicate) {
+    for (Class<?> parameterType : executable.getParameterTypes()) {
+      String parameterName = parameterType.getName();
+      if (!shouldOmitClass(parameterName)
+          && !parameterType.isPrimitive()
+          && !parameterType.equals(String.class) // weird behavior when these are added
```

Comment:
Can you explain what the weird behavior is?  Can you provide a test that exposes the weird behavior?  Then perhaps I can fix the underlying issue.

Reply:
When String class is under test, this error is continuously outputted:
```
Error selecting inputs for operation: java.lang.String.replaceAll : (java.lang.String, java.lang.String, java.lang.String) -> java.lang.String
randoop.main.RandoopBug: getSequencesForType(java.lang.String.replaceAll : (java.lang.String, java.lang.String, java.lang.String) -> java.lang.String, 0, true) neededType=java.lang.Str
ing
        at randoop.generation.ComponentManager.getSequencesForType(ComponentManager.java:186)
        at randoop.generation.ForwardGenerator.selectInputs(ForwardGenerator.java:770)
        at randoop.generation.ForwardGenerator.createNewUniqueSequence(ForwardGenerator.java:457)
        at randoop.generation.ForwardGenerator.step(ForwardGenerator.java:208)
        at randoop.generation.AbstractGenerator.createAndClassifySequences(AbstractGenerator.java:320)
        at randoop.main.GenTests.handle(GenTests.java:514)
        at randoop.main.Main.nonStaticMain(Main.java:71)
        at randoop.main.Main.main(Main.java:31)
```
Seems as error is thrown in this code inside ComponentManager
```java
    if (onlyReceivers && neededType.isNonreceiverType()) {
      throw new RandoopBug(
          String.format(
              "getSequencesForType(%s, %s, %s) neededType=%s",
              operation, i, onlyReceivers, neededType));
    }
```
So I think adding nonreceiver types should be avoided and I will change code to test if type is nonreceiver instead of testing for equality with concrete classes